src/videojs-media-sources.js

```diff
@@ -201,7 +210,18 @@ addTextTrackData = function (sourceHandler, captionArray, metadataArray) {
         }
 
         self.player_ = videojs(video.parentNode);
-        self.trigger(event);
+      });
+
+      // explicitly terminate any WebWorkers that were created
+      // by SourceHandlers
+      this.on('sourceclose', function(event) {
+        this.sourceBuffers.forEach(function(sourceBuffer) {
+          if (sourceBuffer.transmuxer_) {
+            sourceBuffer.transmuxer_.terminate();
```

Comment:
After the player has called dispose on the SourceHandler, the MediaSource will be removed from the Media Element. This will automatically remove SourceBuffers and trigger a `sourceclose` on the MediaSource object which will allow us to remove the web worker.